packages/fxa-settings/src/components/Settings/FlowRecoveryKeyHint/index.test.tsx

```diff
@@ -0,0 +1,147 @@
+    jest.clearAllMocks();
+  });
+
+  it('renders as expected', async () => {
+    renderWithProps();
+
+    screen.getByRole('heading', { level: 1, name: 'Account Recovery Key' });
+    screen.getByRole('progressbar', { name: 'Step 4 of 4.' });
+    screen.getByRole('heading', {
+      level: 2,
+      name: 'Help yourself find your key later',
+    });
+    screen.getByText(
+      'Remember where you saved your key by creating a hint. If you forget your password and canʼt find your key, you might not be able to get your data back.'
+    );
+    screen.getByRole('textbox', { name: 'Enter a hint (optional)' });
+    screen.getByRole('button', { name: 'Finish' });
+  });
+
+  it('emits the expected metrics when the user lands on this step of the flow', () => {
+    renderWithProps();
+    expect(logViewEvent).toBeCalledTimes(1);
+    expect(logViewEvent).toBeCalledWith(viewName, 'hint-step-view');
+  });
+
+  it('saves the hint on submit if the user has entered a valid hint in the text input', async () => {
+    renderWithProps();
+    const hintValue = 'Ye Olde Hint';
+    const textInput = screen.getByRole('textbox', {
+      name: 'Enter a hint (optional)',
+    });
+    const submitButton = screen.getByText('Finish');
+    fireEvent.input(textInput, {
+      target: { value: hintValue },
+    });
+    await waitFor(() => {
+      expect(textInput).toHaveValue(hintValue);
+    });
+    fireEvent.click(submitButton);
+    await waitFor(() => {
+      expect(logViewEvent).toBeCalledWith(viewName, 'hint-submit');
+      expect(account.updateRecoveryKeyHint).toBeCalledWith(hintValue);
+      expect(logViewEvent).toBeCalledWith(viewName, 'hint-success');
+      expect(navigateForward).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  it('displays error tooltip if the hint is too long', async () => {
+    renderWithProps();
+    const hintValueTooLong =
+      'eunqanrhdfiwnzrrigazwjkxywdupiwqlzkneaniwxcnaltdbwlbbtuyozbyxkmqvoiaehcpycjvejabhhyxnvcauxeokprdbbebgmizsphjlbtsscokweckspeqndegwzmlhyrmglbnkmdwvynitqckyubpchplpsxevwtczeuujkznpvtjrchkpbngnmlkqwinzxwswdwinyrgvkvusrpkqcitwznqcsaajqmxtqjlpxxfqbemtxsvolaygntkyvzdtafltvbvvtrxgqfnvdktleuhflkcmwbfracrpbos';
+    const textInput = screen.getByRole('textbox', {
+      name: 'Enter a hint (optional)',
+    });
+    const submitButton = screen.getByText('Finish');
+    fireEvent.input(textInput, {
+      target: { value: hintValueTooLong },
+    });
+    await waitFor(() => {
+      expect(textInput).toHaveValue(hintValueTooLong);
+    });
+    fireEvent.click(submitButton);
+    await waitFor(() => {
+      expect(screen.getByTestId('tooltip')).toHaveTextContent(
+        'The hint must contain fewer than 255 characters.'
+      );
+      expect(account.updateRecoveryKeyHint).not.toBeCalled();
+      expect(logViewEvent).not.toBeCalledWith(viewName, 'hint-submit');
+      expect(navigateForward).not.toBeCalled();
+    });
+  });
+
+  it('logs an error if saving a valid hint failed', () => {
```

Comment:
Do we want to do this in this PR, or are we saving this for a future PR?

Reply:
Sorry for missing this! I'll add it to this PR.